activerecord/lib/active_record/associations/join_dependency.rb

```diff
@@ -272,7 +272,9 @@ def construct(ar_parent, parent, row, rs, seen, model_cache, aliases)
             construct(model, node, row, rs, seen, model_cache, aliases)
           else
             model = construct_model(ar_parent, node, row, model_cache, id, aliases)
-            model.readonly!
+            if node.reflection.scope_for(node.base_klass).values[:readonly]
```

Comment:
`values` dup new Hash. It is better to use `readonly_value`.
https://github.com/rails/rails/blob/v5.0.0/activerecord/lib/active_record/relation.rb#L678-L680
https://github.com/rails/rails/blob/v5.0.0/activerecord/lib/active_record/relation/query_methods.rb#L73-L75